tidy3d/components/mode/data/sim_data.py

```diff
@@ -0,0 +1,119 @@
+"""Mode solver simulation data"""
+
+from __future__ import annotations
+
+from typing import Literal, Tuple
+
+import pydantic.v1 as pd
+
+from ...base import cached_property
+from ...data.monitor_data import ModeSolverData
+from ...data.sim_data import AbstractYeeGridSimulationData
+from ...types import Ax, FieldVal, PlotScale
+from ..simulation import ModeSimulation
+
+# Warning for field intensity at edges over total field intensity larger than this value
+FIELD_DECAY_CUTOFF = 1e-2
+
+
+class ModeSimulationData(AbstractYeeGridSimulationData):
+    """Data associated with a mode solver simulation."""
+
+    simulation: ModeSimulation = pd.Field(
+        ..., title="Mode simulation", description="Mode simulation associated with this data."
+    )
+
+    data_raw: ModeSolverData = pd.Field(
+        ...,
+        title="Raw Data",
+        description="Raw mode data containing the field and effective index " "on unexpanded grid.",
+    )
+
+    data: Tuple[Literal[None], ...] = pd.Field(
```

Comment:
is this needed?

Reply:
sorry, revisiting this after reading more, I think we could consider just `data = Literal[()]`?

Reply:
for some reason this doesn't work:
`E           pydantic.v1.error_wrappers.ValidationError: 2 validation errors for ModeSimulation
E           sources
E             unexpected value; permitted:  (type=value_error.const; given=(); permitted=())
E           monitors
E             unexpected value; permitted:  (type=value_error.const; given=(); permitted=())`